api/src/main/java/io/opencensus/metrics/export/Distribution.java

```diff
@@ -163,6 +132,104 @@ public static Distribution create(
    */
   public abstract List<Bucket> getBuckets();
 
+  /**
+   * The bucket options used to create a histogram for the distribution.
+   *
+   * @since 0.17
+   */
+  @Immutable
+  public abstract static class BucketOptions {
+
+    private BucketOptions() {}
+
+    /**
+     * Returns a ExplicitOptions {@link ExplicitOptions}.
+     *
+     * @param bucketBoundaries the bucket boundaries of a distribution (given explicitly). The
+     *     values must be strictly increasing and should be contains positive values.
+     * @return a {@code ExplicitOptions} {@code BucketOptions}.
+     * @since 0.17
+     */
+    public static BucketOptions explicitOptions(List<Double> bucketBoundaries) {
+      return ExplicitOptions.create(bucketBoundaries);
+    }
+
+    /**
+     * Applies the given match function to the underlying data type.
+     *
+     * @since 0.17
+     */
+    public abstract <T> T match(
+        Function<? super ExplicitOptions, T> explicitFunction,
+        Function<? super BucketOptions, T> defaultFunction);
+
+    /** A Bucket with explicit bounds {@link BucketOptions}. */
+    @AutoValue
+    @Immutable
+    abstract static class ExplicitOptions extends BucketOptions {
```

Comment:
I think this subclass should be made public, otherwise there's no way to call `getBucketBoundaries`.

Reply:
done